src/hexen/g_game.c

```diff
@@ -1292,7 +1325,7 @@ void G_DoReborn(int playernum)
     boolean foundSpot;
     int bestWeapon;
 
-    if (G_CheckDemoStatus())
+    if (!demoextend && G_CheckDemoStatus()) // quit demo unless -demoextend
```

Comment:
Don't leave comments hanging off the end of the line like this. Put above the line if you want to explain what it does.